plugins/BEdita/Core/src/Utility/ProjectModel.php

```diff
@@ -231,12 +231,32 @@ public static function diff(array $project): array
                 $update[$key] = array_values(static::itemsToUpdate($current, $new));
             }
         }
+        self::categoriesToUpdate($update);
 
         return array_filter(
             array_map('array_filter', compact('create', 'update', 'remove'))
         );
     }
 
+    /**
+     * Check if there are categories to update.
+     *
+     * @param array $update Update array
+     * @return void
+     */
+    public static function categoriesToUpdate(array &$update): void
+    {
+        if (!Hash::check($update, 'categories.{n}.name')) {
+            return;
+        }
+        $names = (array)Hash::extract($update['categories'], '{n}.name');
+        $found = TableRegistry::getTableLocator()->get('Categories')->find()->where(['name IN' => $names])->toArray();
```

Comment:
We need to search `name` for object type because the same category name could be used for different object types